src/hivpy/prep.py

```diff
@@ -20,10 +27,226 @@ class PrEPType(IntEnum):
+                                                  COND(col.AGE, op.ge, 15),
+                                                  COND(col.AGE, op.lt, 50)))
+                active_at_risk_pop = pop.get_sub_pop(AND(COND(col.HIV_DIAGNOSED, op.eq, False),
+                                                         OR(COND(col.LAST_STP_DATE, op.ge,
+                                                                 pop.date - timedelta(months=6)),
+                                                            AND(COND(col.LTP_HIV_DIAGNOSED, op.eq, True),
+                                                                COND(col.LTP_ON_ART, op.eq, False)))))
+                # active_at_risk OR (gen_fem AND (risk_informed OR suspect_risk))
+                prep_eligible_pop = pop.get_sub_pop_union(
+                    active_at_risk_pop, pop.get_sub_pop_intersection(
+                        gen_fem_pop, pop.get_sub_pop_union(
+                            self.get_risk_informed_pop(pop, prob_risk_informed_prep), self.get_suspect_risk_pop(pop))))
+            # serodiscordant couples
+            elif self.prep_strategy == 15:
+                gen_ltp_pop = pop.get_sub_pop(AND(COND(col.HIV_DIAGNOSED, op.eq, False),
+                                                  COND(col.LONG_TERM_PARTNER, op.eq, True),
+                                                  COND(col.LTP_HIV_DIAGNOSED, op.eq, False),
+                                                  COND(col.AGE, op.ge, 15),
+                                                  COND(col.AGE, op.lt, 50),
+                                                  OR(COND(col.R_PREP, op.lt, 0.01),  # (alt) risk informed
+                                                     AND(COND(col.R_PREP, op.lt, self.prob_suspect_risk_prep),
+                                                         COND(col.LTP_HIV_STATUS, op.eq, True)))))  # (alt) suspect risk
+                at_risk_ltp_pop = pop.get_sub_pop(AND(COND(col.HIV_DIAGNOSED, op.eq, False),
+                                                      COND(col.LTP_HIV_DIAGNOSED, op.eq, True),
+                                                      COND(col.LTP_ON_ART, op.eq, False)))
+                # at_risk_ltp OR gen_ltp
+                prep_eligible_pop = pop.get_sub_pop_union(at_risk_ltp_pop, gen_ltp_pop)
+            # pregnant and lactating/breastfeeding women
+            elif self.prep_strategy == 16:
+                plw_pop = pop.get_sub_pop(AND(COND(col.HIV_DIAGNOSED, op.eq, False),
+                                              COND(col.SEX, op.eq, SexType.Female),
+                                              OR(COND(col.PREGNANT, op.eq, True),
+                                                 COND(col.BREASTFEEDING, op.eq, True))))
+                active_pop = pop.get_sub_pop(OR(COND(col.LONG_TERM_PARTNER, op.eq, True),
+                                                COND(col.LAST_STP_DATE, op.ge, pop.date - timedelta(months=9))))
```

Comment:
There's a discrepency here with core:
```
	if prep_any_strategy=16 then do;	* Pregnant and breastfeeding women (PLW) - new for MIHPSA Zimbabwe; *JAS Apr2023;
		* Note that there is a component of sexual behaviour in prep eligibility for pregnant and breastfeeding women;
      	if gender=2 and (pregnant=1 or breastfeeding=1) and 
		(newp ge 1 or (epdiag=1 and epart ne 1) or (ep=1 and epart ne 1 and (r < 0.05 or (r < 0.5 and epi=1)))) then prep_any_elig=1; 
		*( newp ge 1 or newp_tm1 ge 1 or newp_tm2 ge 1 or ep=1 ) then prep_any_elig=1; 	* replaced with code to match AGYW above Jan24;
	end;
```

The implementation which uses the previous two timesteps has been commented out; is there a reason why we are using this verion here?

Reply:
The current implementation is based on the eligibility code in `hiv_synthesis_laa_s.sas` on the `long_acting_treatment` branch:
```
	if prep_any_strategy=16 then do;	* Pregnant and breastfeeding women (PLW) - new for MIHPSA Zimbabwe; *JAS Apr2023;
		* Note that there is a component of sexual behaviour in prep eligibility for pregnant and breastfeeding women;
      	if gender=2 and (pregnant=1 or breastfeeding=1) and ( newp ge 1 or newp_tm1 ge 1 or newp_tm2 ge 1 or ep=1 ) then prep_any_elig=1; 
	end;
```
I've been using this branch because it contains some PrEP updates, such as the split of injectable PrEP into Cab and Len.

Reply:
Git blame puts these lines in both branches at about 5 months ago... @andrew-phillips-1 @jennifersmith203 do either of you know which version of this strategy we should be using in the python code? (As you two come up as the ones who have most recently edited this section!)